### PR TITLE
PS-5686: improve innodb_redo_log_encrypt validation and formatting (5.7)

### PR DESCRIPTION
**Pull request by dutow**
* Variable no longer accepts 2 and 3 as valid values
* Value is dispayed in upcase as with other variables

### CODE REVIEW

File: storage/innobase/handler/ha_innodb.cc [hunk 1]
```diff
@@ -946,6 +946,17 @@ innodb_encrypt_tables_validate(
 						for update function */
 	struct st_mysql_value*		value);	/*!< in: incoming string */
 
+static
+int
+innodb_redo_log_encrypt_validate(
+/*==================================*/
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc]: New InnoDB header comment style in new code please

File: storage/innobase/handler/ha_innodb.cc [hunk 2]
```diff
@@ -23996,3 +24007,59 @@ innodb_encrypt_tables_validate(
 
 	return 0;
 }
+
+static int
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc]: New InnoDB header comment style in new code please

File: storage/innobase/handler/ha_innodb.cc [hunk 3]
```diff
@@ -23996,3 +24007,59 @@ innodb_encrypt_tables_validate(
...
+    THD *		     thd,  /*!< in: thread handle */
+    struct st_mysql_sys_var *var,  /*!< in: pointer to system
+                                                  variable */
+    void *		     save, /*!< out: immediate result
+                                  for update function */
+    struct st_mysql_value *  value)  /*!< in: incoming string */
+{
+	const char *redo_log_encrypt_input;
+	char	buff[STRING_BUFFER_USUAL_SIZE];
+	int	 len = sizeof(buff);
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc]: Formatting

File: storage/innobase/handler/ha_innodb.cc [hunk 4]
```diff
@@ -23996,3 +24007,59 @@ innodb_encrypt_tables_validate(
...
+
+	ut_a(save != NULL);
+	ut_a(value != NULL);
+
+	redo_log_encrypt_input = value->val_str(value, buff, &len);
+
+	bool legit_value = false;
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc:24037]: Is this new code or copy-pasted from somewhere? If yes, from where?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by laurynas-biveinis** [comment]
Please also fix the JIRA issue number in the commit message

File: storage/innobase/handler/ha_innodb.cc (hunk 4 above)
Comment (dutow) [reply storage/innobase/handler/ha_innodb.cc:24037]: I copied the innodb_encrypt_tables validation and changed it to match redo_log_encrypt.

File: storage/innobase/handler/ha_innodb.cc [hunk 5]
```diff
@@ -23996,3 +24009,62 @@ innodb_encrypt_tables_validate(
...
+		use = 1;
+		legit_value = true;
+	}
+
+	if (innobase_strcasecmp(redo_log_encrypt_input, "true") == 0) {
+		use = 1;
+		legit_value = true;
+	}
+
+	if (legit_value == false)
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc]: (Github is crashing, this is my full review)
```suggestion
	if (!legit_value)
```

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by dutow** [comment]
Rebased on 5.7 after ps-5565 merge, no other change. 

https://ps57.cd.percona.com/job/percona-server-5.7-param/96/

**Comment by commitcheck** [comment]
Hi @laurynas-biveinis, I noticed your comment about adding a JIRA issue number to the commit message. I build a Github app that will automate that for you. Hopefully this will save you time on future PRs. It's free to use on OSS projects -> https://github.com/apps/commitcheck-app

**Pull request merged**